components/ArtistSetting/AddToOrgButton.tsx

```diff
@@ -0,0 +1,101 @@
+"use client";
+
+import { useState, useRef, useEffect } from "react";
+import { Building2, ChevronDown, Loader2 } from "lucide-react";
+import { cn } from "@/lib/utils";
+import { buttonPatterns } from "@/lib/styles/patterns";
+import useUserOrganizations from "@/hooks/useUserOrganizations";
+import { useArtistProvider } from "@/providers/ArtistProvider";
+import { useOrganization } from "@/providers/OrganizationProvider";
+
+interface AddToOrgButtonProps {
+  artistId: string;
+}
+
+const AddToOrgButton = ({ artistId }: AddToOrgButtonProps) => {
+  const { data: organizations } = useUserOrganizations();
+  const { toggleSettingModal } = useArtistProvider();
+  const { setSelectedOrgId } = useOrganization();
+  const [addingToOrg, setAddingToOrg] = useState<string | null>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  // Close dropdown when clicking outside
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+        setIsOpen(false);
+      }
+    };
+    if (isOpen) {
+      document.addEventListener("mousedown", handleClickOutside);
+    }
+    return () => document.removeEventListener("mousedown", handleClickOutside);
+  }, [isOpen]);
+
+  // Don't render if user has no orgs
+  if (!organizations || organizations.length === 0) {
+    return null;
+  }
+
+  const handleAddToOrg = async (orgId: string) => {
+    setAddingToOrg(orgId);
+    try {
+      const response = await fetch("/api/artist/add-to-org", {
+        method: "POST",
+        body: JSON.stringify({
+          artistId,
+          organizationId: orgId,
+        }),
+        headers: { "Content-Type": "application/json" },
+      });
+
+      if (response.ok) {
+        // Close modal and switch to the org
+        toggleSettingModal();
+        setSelectedOrgId(orgId);
+        // Artists will auto-refresh due to org change
+      }
+    } finally {
+      setAddingToOrg(null);
+    }
+  };
```

Comment:
Single Responsibility Principle
- actual: fetch / handler is defined directly in a react component.
- required: new hook for `hooks/useAddArtistToOrganization.ts`